### PR TITLE
0x tranqui/edit members foundation

### PR DESCRIPTION
**Pull request by 0xTranqui**

### CODE REVIEW

File: apps/site/app/(base)/page.tsx
Comment (salieflewis) [inline apps/site/app/(base)/page.tsx:1]: Did these changes get reverted when dealing with merge conflicts?

File: apps/site/components/client/ChannelSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+  });
+
+  type Roles = {
+    rid: bigint;
+    username: string;
+    startingRole: bigint;
+    newRole: bigint | null;
+  };
+  const [roles, setRoles] = useState<Roles[]>([]);
+  console.log("roles state: ", roles);
```
Comment (salieflewis) [inline apps/site/components/client/ChannelSettings.tsx:94]: Rm.

File: apps/site/components/client/ChannelSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+    (role) => role.newRole || BigInt(0)
+  ); // default to BigInt(0) if newRole is null);
+
+  /* 
+  *
+    username input validation
+  *
+  */
+
+  const [isValidUser, setIsValidUser] = useState(0); // 0 null state, 1 = valid, 2 = invalid, 3 = already added
```
Comment (salieflewis) [inline apps/site/components/client/ChannelSettings.tsx:115]: `const [isValidUser, setIsValidUser] = useState<0 | 1 | 2 | 3>(0); // 0 = null state, 1 = valid, 2 = invalid, 3 = already added`

File: apps/site/components/client/ChannelSettings.tsx [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+          setIsCheckingUsername(false);
+        });
+      }
+      return () => {
+        setValidationComplete(false);
+      };
+    }
+  }, [validationComplete, watchUsername]);
+
+  /* NOTE: probably should add a check to ensure that number of admins/members works nice in UI */
```
Comment (salieflewis) [inline apps/site/components/client/ChannelSettings.tsx:154]: Can we rm?

Comment (0xTranqui) [reply apps/site/components/client/ChannelSettings.tsx:154]: not yet. still something to keep an eye on

**Review by salieflewis — APPROVED** [review]
Excited for this!

**Pull request merged**